### PR TITLE
files and folders cannot end with a '.' on windows

### LINKED ISSUE #117426: Possibility to create a file that cannot be deleted, rename or moved.
- VS Code Version: 1.53.2 (user setup)
- OS Version: Windows_NT x64 10.0.19041

Steps to Reproduce:

1.  Put a dot at the end of the file name. For example - "AnyName."

This file can't even delete "Unlocker".
This file bypasses the Windows restriction. Which prohibits the creation of such files.
I assume that using "Process Monitor" can understand exactly how visual studio does this and use it.
It shouldn't work like that.

osbTVqCHcw

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by nrayburn-tech**
Files and folders cannot end with a `.` on windows.  This PR throws an error when a file/folder is created that has a trailing `.`.

I created the test based off of a similar one in the file, but am on a mac and cannot actually test it.  It may need to be updated/validated on a windows computer.

The windows file explorer behavior is slightly different.  It just removes the trailing `.`, so if that is preferred I can modify the PR to do that.  I would just need a pointer to where I should be modifying the resource URI to remove the trailing `.`.

This PR fixes #117426

### CODE REVIEW

**Comment by isidorn** [comment]
Change in the fileService, leaving up to @bpasero to review

**Review by bpasero — CHANGES_REQUESTED** [review]
The change cannot be made on the level of `IFileService´ but has to be inside the file system provider (i.e. `diskFileSystemProvider`) because the actual file system provider that is being used in the end may not be OS specific (could be backed by a database for example).

File: src/vs/platform/files/common/files.ts
Comment (nrayburn-tech) [inline src/vs/platform/files/common/files.ts:469]: Not entirely sure about the naming for this.

File: src/vs/platform/files/test/electron-browser/diskFileService.test.ts
Comment (nrayburn-tech) [inline src/vs/platform/files/test/electron-browser/diskFileService.test.ts:1683]: I didn't see any other test in this file using `assert.rejects`, but I think it is the correct way to go.  If not, I can use another method.

**Comment by bpasero** [comment]
I just realised that the explorer seems to have a bug and commented in https://github.com/microsoft/vscode/issues/117426#issuecomment-834171811. I would prefer a fix in the explorer and not on the file service to be honest.

**Comment by nrayburn-tech** [comment]
I’ll wait for a reply on the issue, and if a PR is still needed for the explorer I’ll re-open this with the changes.

**Pull request closed without merging**
